FindStandardFortran.cmake

```diff
@@ -0,0 +1,41 @@
+#
+# This module will locate the Fortran standard libraries so that a Fortran
+# object or library can be linked against using a C/C++ compiler.
+#
+# After calling this module the following will be set:
+#    STANDARDFORTRAN_LIBRARIES : A list of libraries to link against
+#    StandardFortran_FOUND     : True if we found the standard Fortran libraries
+#
+# Implemenation note.  CMake provides a variable
+# CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES which is a list of all libraries a
+# compiler implicitly links against.  Unfortunately, at least for GNU, this
+# list includes a lot of extra libraries that we don't necessarilly want to
+# link against (including both static and shared versions of libgcc).  This is
+# why we've hardcoded the list per compiler.
+#
+enable_language(Fortran)
+include(FindPackageHandleStandardArgs)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+    set(STANDARDFORTRAN_LIBS gfortran)
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+    set(STANDARDFORTRAN_LIBS ifcore)
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Flang")
+    set(STANDARDFORTRAN_LIBS flang flangrti pgmath)
+else()
+    message(FATAL_ERROR "${CMAKE_Fortran_COMPILER_ID} not supported yet.")
```

Comment:
This need to be more descriptive in that its specifically in conjunction with this module